backend/src/user/user.service.ts

```diff
@@ -36,7 +36,13 @@ export class UserService {
       throw new HttpException('User not found.', HttpStatus.NOT_FOUND);
     } else {
       deletedUser = users[index];
-      users.slice(index, 1);
+      const newUsers: User[] = [];
+      for(let i = 0; i < users.length; i++) {
+        if(index !== i) {
```

Comment:
You can simplify this with `deletedUser = user.splice(index, 1)` so basically almost like you already had, but with a **p** (yes there really are `slice` and `splice`. Don't get me started on `substring` and `substr`). See <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice>

Reply:
Oh didn't notice that.